substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1283,10 +1285,18 @@ pub mod pallet {
 			payee: RewardDestination<T::AccountId>,
 		) -> DispatchResult {
 			let controller = ensure_signed(origin)?;
-			let ledger = Self::ledger(Controller(controller))?;
+			let ledger = Self::ledger(Controller(controller.clone()))?;
+
+			let controller_deprecated = || -> RewardDestination<T::AccountId> {
```

Comment:
Not sure about this, but could this be replaced by annotating the `ensure!` with `#[allow(deprecated)]`?

Reply:
The annotation does not work if applied to `ensure!`:

```
the built-in attribute `allow` will be ignored, since it's applied to the macro invocation `ensure`rustc(unused_attributes)
```

Reply:
We can get rid of this with the suggestion below.